src/psyclone/psyir/symbols/datatypes.py

```diff
@@ -870,10 +897,21 @@ class StructureType(DataType):
     SymbolTable functionality then this decision could be revisited.
 
     '''
-    # Each member of a StructureType is represented by a ComponentType
-    # (named tuple).
-    ComponentType = namedtuple("ComponentType", [
-        "name", "datatype", "visibility", "initial_value"])
+    @dataclass(frozen=True)
+    class ComponentType:
+        '''
+        Represents a member of a StructureType.
+
+        :param name: the name of the member.
+        :param datatype: the type of the member.
+        :param visibility: whether this member is public or private.
+        :param initial_value: the initial value of this member (if any).
+        :type initial_value: Optional[:py:class:`psyclone.psyir.nodes.Node`]
+        '''
+        name: str
+        datatype: DataType | DataTypeSymbol
```

Comment:
This make CI fail:
```
  File "/opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/site-packages/psyclone/psyir/symbols/datatypes.py", line 912, in ComponentType
    datatype: DataType | DataTypeSymbol
TypeError: unsupported operand type(s) for |: 'ABCMeta' and 'type'
```
I think you will need to use `Union` importing it from `typing`.

Reply:
Thanks yes. It turns out I hit 'ready for review' a bit too soon. There was another, similar problem in the new handler in conf.py that I've also had to fix.